pkg/kubelet/kubelet_test.go

```diff
@@ -2793,68 +2817,109 @@ func TestHandlePodResourcesResize(t *testing.T) {
-			} else {
-				require.NoError(t, kubelet.statusManager.SetPodAllocation(newPod))
-			}
+				if isSidecarContainer {
+					newPod.Spec.InitContainers[0].Resources.Requests = tt.newRequests
+				} else {
+					newPod.Spec.Containers[0].Resources.Requests = tt.newRequests
+				}
 
-			podStatus := &kubecontainer.PodStatus{
-				ID:                originalPod.UID,
-				Name:              originalPod.Name,
-				Namespace:         originalPod.Namespace,
-				ContainerStatuses: make([]*kubecontainer.Status, len(originalPod.Spec.Containers)),
-			}
-			for i, c := range originalPod.Spec.Containers {
-				podStatus.ContainerStatuses[i] = &kubecontainer.Status{
-					Name:  c.Name,
-					State: kubecontainer.ContainerStateRunning,
-					Resources: &kubecontainer.ContainerResources{
-						CPURequest:  c.Resources.Requests.Cpu(),
-						CPULimit:    c.Resources.Limits.Cpu(),
-						MemoryLimit: c.Resources.Limits.Memory(),
-					},
+				if !tt.newRequestsAllocated {
+					require.NoError(t, kubelet.statusManager.SetPodAllocation(originalPod))
+				} else {
+					require.NoError(t, kubelet.statusManager.SetPodAllocation(newPod))
 				}
-			}
 
-			now := kubelet.clock.Now()
-			// Put the container in backoff so we can confirm backoff is reset.
-			backoffKey := kuberuntime.GetStableKey(originalPod, &originalPod.Spec.Containers[0])
-			kubelet.backOff.Next(backoffKey, now)
+				podStatus := &kubecontainer.PodStatus{
+					ID:        originalPod.UID,
+					Name:      originalPod.Name,
+					Namespace: originalPod.Namespace,
+				}
 
-			updatedPod, err := kubelet.handlePodResourcesResize(newPod, podStatus)
-			require.NoError(t, err)
-			assert.Equal(t, tt.expectedAllocations, updatedPod.Spec.Containers[0].Resources.Requests, "updated pod spec resources")
+				setContainerStatus := func(podStatus *kubecontainer.PodStatus, c *v1.Container, idx int) {
+					podStatus.ContainerStatuses[idx] = &kubecontainer.Status{
+						Name:  c.Name,
+						State: kubecontainer.ContainerStateRunning,
+						Resources: &kubecontainer.ContainerResources{
+							CPURequest:  c.Resources.Requests.Cpu(),
+							CPULimit:    c.Resources.Limits.Cpu(),
+							MemoryLimit: c.Resources.Limits.Memory(),
+						},
+					}
+				}
 
-			alloc, found := kubelet.statusManager.GetContainerResourceAllocation(string(newPod.UID), newPod.Spec.Containers[0].Name)
-			require.True(t, found, "container allocation")
-			assert.Equal(t, tt.expectedAllocations, alloc.Requests, "stored container allocation")
+				if isSidecarContainer {
+					podStatus.ContainerStatuses = make([]*kubecontainer.Status, len(originalPod.Spec.InitContainers))
+					for i, c := range originalPod.Spec.InitContainers {
+						setContainerStatus(podStatus, &c, i)
+					}
+				} else {
+					podStatus.ContainerStatuses = make([]*kubecontainer.Status, len(originalPod.Spec.Containers))
+					for i, c := range originalPod.Spec.Containers {
+						setContainerStatus(podStatus, &c, i)
+					}
+				}
```

Comment:
Just add both unconditionally.
```suggestion
				podStatus.ContainerStatuses = make([]kubecontainer.Status, len(originalPod.Spec.Containers) + len(originalPod.Spec.InitContainers))
				for i, c := range originalPod.Spec.InitContainers {
					setContainerStatus(podStatus, &c, i)
				}
				for i, c := range originalPod.Spec.Containers {
					setContainerStatus(podStatus, &c, i + len(originalPod.Spec.InitContainers))
				}
```